src/modules/auth/screens/sign-up.tsx

```diff
@@ -0,0 +1,167 @@
+import React, {useRef, useState} from 'react';
+import {
+  Alert,
+  Image,
+  Keyboard,
+  KeyboardAvoidingView,
+  Platform,
+  StyleSheet,
+  Text,
+  TextInput,
+  useWindowDimensions,
+} from 'react-native';
+import {CustomButton} from '../../../components/custom-button';
+import {CustomInput} from '../../../components/custom-input';
+import {useAuth} from '../hooks/use-auth';
+import DatePicker from 'react-native-datepicker';
+import Logo from '../../../assets/images/logo.png';
+
+export function SignUpScreen(this: {
```

Comment:
Recomendo não usar o termo `this` porque é uma palavra reservada da linguagem, e ela já vem pré-definida tanto dentro de classes quanto dentro de funções. O melhor que costumamos usar pra props é `props` mesmo.